cmd/algorand-indexer/daemon.go

```diff
@@ -233,39 +230,21 @@ var daemonCmd = &cobra.Command{
 
 				// Initial import if needed.
 				genesisReader := importer.GetGenesisFile(genesisJSONPath, bot.Algod(), logger)
-				_, err := importer.EnsureInitialImport(db, genesisReader, logger)
+				genesis, err := readGenesis(genesisReader)
+				maybeFail(err, "Error reading genesis file")
+
+				_, err = importer.EnsureInitialImport(db, genesis, logger)
 				maybeFail(err, "importer.EnsureInitialImport() error")
 
 				// sync local ledger
 				nextDBRound, err := db.GetNextRoundToAccount()
 				maybeFail(err, "Error getting DB round")
-				if nextDBRound > 0 {
-					if catchpoint != "" {
-						round, _, err := ledgercore.ParseCatchpointLabel(catchpoint)
-						if err != nil {
-							maybeFail(err, "catchpoint error")
-						}
-						if uint64(round) >= nextDBRound {
-							logger.Warnf("round for given catchpoint is ahead of db round. skip fast catchup")
-						} else {
-							err = localledger.RunMigrationFastCatchup(logging.NewLogger(), catchpoint, &opts)
-							maybeFail(err, "Error running ledger migration in fast catchup mode")
-						}
-
-					}
-					err = localledger.RunMigrationSimple(nextDBRound-1, &opts)
-					maybeFail(err, "Error running ledger migration")
-				}
```

Comment:
Moved into `MakeProcessorWithLedgerInit`. This keeps all the logic contained in the block processor and should make it easier to merge with Eric's refactoring.